airflow-core/src/airflow/ui/src/pages/Events/EventsFilters.tsx

```diff
@@ -0,0 +1,226 @@
+
+  const handleInputChange = useCallback(
+    (paramName: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+
+      if (value) {
+        searchParams.set(paramName, value);
+      } else {
+        searchParams.delete(paramName);
+      }
+      resetPagination();
+      setSearchParams(searchParams);
+    },
+    [resetPagination, searchParams, setSearchParams],
+  );
+
+  const handleDateTimeChange = useCallback(
+    (paramName: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+
+      if (value) {
+        // Convert datetime-local value to ISO string
+        try {
+          const date = new Date(value);
+
+          if (!isNaN(date.getTime())) {
+            searchParams.set(paramName, date.toISOString());
+          }
+        } catch {
+          // Invalid date, remove the parameter
+          searchParams.delete(paramName);
+        }
+      } else {
+        searchParams.delete(paramName);
+      }
+      resetPagination();
+      setSearchParams(searchParams);
+    },
+    [resetPagination, searchParams, setSearchParams],
+  );
+
+  const handleClearFilters = useCallback(() => {
+    searchParams.delete(AFTER_PARAM);
+    searchParams.delete(BEFORE_PARAM);
+    searchParams.delete(DAG_ID_PARAM);
+    searchParams.delete(EVENT_TYPE_PARAM);
+    searchParams.delete(MAP_INDEX_PARAM);
+    searchParams.delete(RUN_ID_PARAM);
+    searchParams.delete(TASK_ID_PARAM);
+    searchParams.delete(TRY_NUMBER_PARAM);
+    searchParams.delete(USER_PARAM);
+
+    resetPagination();
+    setSearchParams(searchParams);
+  }, [resetPagination, searchParams, setSearchParams]);
+
+  const filterCount = getFilterCount({
+    after: afterFilter,
+    before: beforeFilter,
+    dagId: dagIdFilter,
+    eventType: eventTypeFilter,
+    mapIndex: mapIndexFilter,
+    runId: runIdFilter,
+    taskId: taskIdFilter,
+    tryNumber: tryNumberFilter,
+    user: userFilter,
+  });
+
+  return (
+    <HStack justifyContent="space-between">
+      <HStack flexWrap="wrap" gap={4}>
+        {/* Timestamp Range Filters */}
+        <Input
+          onChange={handleDateTimeChange(AFTER_PARAM)}
+          placeholder={translate("auditLog.filters.startDateTime")}
+          size="sm"
+          type="datetime-local"
+          value={formatDateTimeLocalValue(afterFilter)}
+          width="200px"
+        />
+        <Input
+          onChange={handleDateTimeChange(BEFORE_PARAM)}
+          placeholder={translate("auditLog.filters.endDateTime")}
+          size="sm"
+          type="datetime-local"
+          value={formatDateTimeLocalValue(beforeFilter)}
+          width="200px"
+        />
+
+        {/* Event Type Filter */}
+        <Input
+          onChange={handleInputChange(EVENT_TYPE_PARAM)}
+          placeholder={translate("auditLog.filters.eventType")}
+          size="sm"
+          value={eventTypeFilter}
+          width="150px"
+        />
```

Comment:
We will need to maybe create a `<DebounceInput`  component or simply add a debounce to all inputs, similarly to the `const SearchBar`.

Otherwise this will trigger a backend request on each keystroke, which can make a lot of unecessary request.